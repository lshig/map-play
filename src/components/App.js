import React from 'react';
import EventMap from './EventMap';
import SEO from './SEO';

export default function App() {
  return (
    <section>
      <SEO />
      <a
        href="https://lizshigetoshi.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h1 id="lizshigetoshi">Liz Shigetoshi</h1>
      </a>
      <EventMap />
    </section>
  );
}
