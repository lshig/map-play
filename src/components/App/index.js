'use strict'
import React from 'react'
import Interactive from '../Interactive'
const App = () => (
  <section>
    <a href='http://lizshigetoshi.com' target='_blank'>
      <h1 id='lizshigetoshi'>Liz Shigetoshi</h1>
    </a>
    <Interactive />
  </section>
)
export default App
