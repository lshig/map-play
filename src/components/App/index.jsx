import React from 'react'
import MapBox from '../MapBox'
export default function App () {
  return (
    <section>
      <a href="http://lizshigetoshi.com" target="_blank">
        <h1 id="lizshigetoshi">Liz Shigetoshi</h1>
      </a>
      <MapBox />
    </section>
  )
}
