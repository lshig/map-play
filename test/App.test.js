import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../src/components/App';

jest.mock('mapbox-gl', () => {
  return {
    accessToken: 'foobar',
    Map: jest.fn(),
    Marker: jest.fn(() => {
      return {
        setLngLat: jest.fn(() => {
          return {
            addTo: jest.fn(),
            setPopup: jest.fn(() => {
              return {
                addTo: jest.fn()
              };
            })
          };
        })
      };
    }),
    Popup: jest.fn(() => {
      return {
        setDOMContent: jest.fn()
      };
    })
  };
});

describe('App', () => {
  describe('basic', () => {
    const wrapper = shallow(<App />);

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
