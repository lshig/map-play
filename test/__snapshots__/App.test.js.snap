// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App basic renders 1`] = `
<section>
  <a
    href="https://lizshigetoshi.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <h1
      id="lizshigetoshi"
    >
      Liz Shigetoshi
    </h1>
  </a>
  <MapboxMap />
</section>
`;
